store/sqlstore/post_store.go

```diff
@@ -1970,7 +1970,7 @@ func (s *SqlPostStore) GetPostsCreatedAt(channelId string, time int64) ([]*model
 func (s *SqlPostStore) GetPostsByIds(postIds []string) ([]*model.Post, error) {
 	keys, params := MapStringsToQueryParams(postIds, "Post")
 
-	query := `SELECT p.*, (SELECT count(Posts.Id) FROM Posts WHERE Posts.RootId = (CASE WHEN p.RootId = '' THEN p.Id ELSE p.RootId END) AND Posts.DeleteAt = 0) as ReplyCount FROM Posts p WHERE p.Id IN ` + keys + ` ORDER BY CreateAt DESC`
+	query := `SELECT p.*, (SELECT count(Posts.Id) FROM Posts WHERE Posts.RootId = (CASE WHEN p.RootId = '' THEN p.Id ELSE p.RootId END) AND Posts.DeleteAt = 0) as ReplyCount FROM Posts p WHERE p.Id IN ` + keys + ` ORDER BY CreateAt DESC LIMIT 1000`
```

Comment:
We should not hardcode limits like this in the SQL query. It breaks other callers of this API. You might query for 1001 IDs, but get only 1000 posts.

Instead, we should enforce the limitation on the API layer. In the API layer, please check for the number of post ids, and if they are more than 1000, throw an error to enforce proper usage rather than silently truncating posts.

Reply:
Enforced on the api layer now